client/home/lib/myteam/components/hometeamteammates/title.coffee

```diff
@@ -0,0 +1,91 @@
+kd    = require 'kd'
+React = require 'app/react'
+TeamFlux       = require 'app/flux/teams'
+ButtonWithMenu = require 'app/components/buttonwithmenu'
+capitalizeFirstLetter = require 'app/util/capitalizefirstletter'
+
+module.exports = class HomeTeamTeamMatesTitle extends React.Component
+
+  constructor: (props) ->
+
+    super props
+
+    @state = {
+      isMenuOpen: no
+      filterKey: 'all'
+    }
+
+    @filterValue = 'See All'
+
+
+  onClickFilter: (event) ->
+
+    kd.utils.stopDOMEvent event
+    @setState { isMenuOpen: yes }
+    
+      
+  onClickTitle: (event) ->
+    if hash = event.target?.hash
+      window.location.replace hash
+  
+    
+  handleFilterChange: (key, value) ->
+    @props.filterValue = value
+    @setState { isMenuOpen: no }
+    @filterValue = value
+    TeamFlux.actions.setRoleFilterKey key
+
+
+  getFilterValue: ->
+    return @filterValue
+    
+    
+  getMenuItems: ->
+
+    items = []
+    items.push { title: 'See All', key: 'all', onClick: @handleFilterChange.bind(this, 'all', 'See All') }
+    items.push { title: 'Only Admins', key: 'admin', onClick: @handleFilterChange.bind(this, 'admin', 'Only Admins') }
+    items.push { title: 'Only Members', key: 'member', onClick: @handleFilterChange.bind(this, 'member', 'Only Members') }
+    items.push { title: 'Only Invited', key: 'invite', onClick: @handleFilterChange.bind(this, 'invited', 'Only Invited') }
+    items.push { title: 'Only Disabled Users', key: 'disable', onClick: @handleFilterChange.bind(this, 'disabled', 'Only Disabled Users') }
+
+    return items
+
+
+  render: ->
+
+    <div className="teammate-header">
+      <TeammateTitle onClick={@onClickTitle.bind(this)} />
```

Comment:
We have a method named `bound` in our [base ReactComponent](https://github.com/koding/koding/blob/master/client/app/lib/react/component.coffee#L18-L29) which can be used instead of this kind of manual bindings:

before:
```coffee
<SomeComponent onClick={@onClick.bind this} />
```
after:
```coffee
<SomeComponent onClick={@bound 'onClick'} />
```

One thing to note: You can only use `bound` method if you don't want to pass any arguments by yourself, so:
```coffee
# this one will pass `foo` as first argument to the `onClick` method.
@onClick.bind this, 'foo'
# we don't have a way to pass `foo` to the `onClick` method with `@bound`
# NOT GONNA WORK!!
@bound('onClick', 'foo') == @bound('onClick')
# => true
```